src/components/time-picker/time-picker.tsx

```diff
@@ -0,0 +1,374 @@
+                            placeholder="00"
+                            disabled={disabled}
+                            name={name}
+                            onMouseDown={this.onInputMouseDown}
+                            onChange={this.onInputChange}
+                            onFocus={this.onInputFocus}
+                            onBlur={this.onBlur}
+                            onKeyDown={this.onKeyDown}
+                        />
+                    </div>
+                )}
+                {format === 'ampm' &&
+                    <div
+                        data-automation-id="TIME_PICKER_AMPM"
+                        className="ampm"
+                        ref={elem => this.segments.ampm = elem}
+                        tabIndex={disabled || isTouch ? -1 : 0}
+                        children={ampmLabels[ampm]}
+                        onMouseDown={this.onAmpmMouseDown}
+                        onFocus={this.onAmpmFocus}
+                        onBlur={this.onBlur}
+                        onKeyDown={this.onKeyDown}
+                    />
+                }
+                {placeholder && !valueSet &&
+                    <div
+                        data-automation-id="TIME_PICKER_PLACEHOLDER"
+                        className="placeholder"
+                        children={placeholder}
+                        onClick={this.onPlaceholderClick}
+                    />
+                }
+                {!isTouch && !disabled && valueSet &&
+                    <Stepper
+                        className="stepper"
+                        onUp={this.onStepperUp}
+                        onDown={this.onStepperDown}
+                    />
+                }
+                {isTouch &&
+                    <label className="label">
+                        <input
+                            className="native-input"
+                            type="time"
+                            name={name}
+                            aria-label={label}
+                            value={this.getValue()}
+                            disabled={disabled}
+                            onFocus={this.onNantiveInputFocus}
+                            onBlur={this.onBlur}
+                            onChange={this.onNativeInputChange}
+                        />
+                    </label>
+                }
+            </div>
+        );
+    }
+
+    private getValue(): string {
+        const {hh = 0, mm = 0, ampm} = this.state;
+        const hhString = formatTimeChunk(to24(Number(hh), ampm));
+        const mmString = formatTimeChunk(mm);
+        return `${hhString}:${mmString}`;
+    }
+
+    private select(segment: Segment) {
+        const input = this.segments[segment]!;
+        this.setState({
+            currentSegment: segment
+        }, () => {
+            input.focus();
+            if (isTimeSegment(segment)) {
+                (input as HTMLInputElement).select();
```

Comment:
no, i can not avoid this, because there is no explisit coleration between `segment` (or `currentSegment`) and `input`